doc/source/tutorial/stats/sampling_tdr.rst

```diff
@@ -0,0 +1,190 @@
+  kept above 0.99 but that can be changed by passing a ``max_sqhratio``
+  parameter.
+* 'hat_area': area below the hat for the generator.
+* 'squeeze_area': area below the squeeze for the generator.
+
+    >>> rng.sqhratio
+    0.9947024204884917
+    >>> rng.hat_area
+    2.510253139791547
+    >>> rng.squeeze_area
+    2.4969548741894876
+    >>> rng.sqhratio == rng.squeeze_area / rng.hat_area
+    True
+
+To increase ``sqhratio``, pass ``max_sqhratio``:
+
+    >>> rng = TransformedDensityRejection(dist, max_sqhratio=0.999,
+    ...                                   max_intervals=1000, seed=urng)
+    >>> rng.sqhratio
+    0.999364900465214
+
+Note that we need to increase the ``max_intervals`` parameter when we want
+a higher ``sqhratio``. This is because more construction points are required
+to fit the distribution more tightly.
+
+Let's see how this affects the callbacks to the PDF method of the
+distribution:
+
+    >>> class StandardNormal:
+    ...     def __init__(self):
+    ...         self.callbacks = 0
+    ...     def pdf(self, x):
+    ...         self.callbacks += 1
+    ...         return np.exp(-0.5 * x*x)
+    ...     def dpdf(self, x):
+    ...         return -x * np.exp(-0.5 * x*x)
+    ... 
+    >>> dist1 = StandardNormal()
+    >>> urng = np.random.default_rng(0x7344aca86f1dafd8820be998b639551c)
+    >>> rng = TransformedDensityRejection(dist1, seed=urng)
+    >>> dist1.callbacks  # evaluations during setup
+    139
+    >>> dist1.callbacks = 0  # don't consider evaluations during setup
+    >>> rvs = rng.rvs(100000)
+    >>> dist1.callbacks  # evaluations during sampling
+    551
+    >>> dist2 = StandardNormal()
+    >>> # use the same stream of uniform random numbers
+    >>> urng = np.random.default_rng(0x7344aca86f1dafd8820be998b639551c)
+    >>> rng = TransformedDensityRejection(dist2, max_sqhratio=0.999,
+    ...                                   max_intervals=1000, seed=urng)
+    >>> dist2.callbacks  # evaluations during setup
+    467
+    >>> dist2.callbacks = 0  # don't consider evaluations during setup
+    >>> rvs = rng.rvs(100000)
+    >>> dist2.callbacks  # evaluations during sampling
+    63
+
+As we can see, far fewer PDF evaluations are required during sampling when
+we increase the ``sqhratio``. Also, notice that this comes at the cost of
+increased PDF evaluations during setup.
+
+For densities with modes not close to 0, it is suggested to set either the
+mode or the center of the distribution by passing ``mode`` or ``center``
+parameters. The latter is the approximate location of the mode or the mean
+of the distribution. This location provides some information about the main
+part of the PDF and is used to avoid numerical problems.
+
+    >>> # mode = 0 for our distribution
+    >>> # if exact mode is not available, pass 'center' parameter instead
+    >>> rng = TransformedDensityRejection(dist, mode=0.)
+
+By default, the method uses 30 construction points to construct the hat.
```

Comment:
I am wondering if we should invite people to use QMC if they want to specify their set of points. What do you think?

Reply:
> I am wondering if we should invite people to use QMC if they want to specify their set of points. What do you think?

I cannot find any literature that recommends doing that. I think if we were to write something by ourselves, we should make it absolutely sure that it is correct usage. Why do you think QMC samples would make good construction points as opposed to, say, partition the domain where the PDF is either strictly concave or strictly convex (as recommended [here])? If there is a good reason to do that, can we add a reference to it?

[here]: https://link.springer.com/content/pdf/10.1007/s11222-011-9306-4.pdf

Reply:
Correct me if I am wrong 😅 Here we are using 30 points to estimate a PDF right? Because QMC is known to reduce the error on the variance and mean to estimate a PDF compare to MC. You simply cover the space more with QMC which directly correlate to a better approximation. Of course, if you have any prior on extrema, you should inform the fit with such points.

Reply:
> Correct me if I am wrong sweat_smile Here we are using 30 points to estimate a PDF right? Because QMC is known to reduce the error on the variance and mean to estimate a PDF compare to MC. You simply cover the space more with QMC which directly correlate to a better approximation. Of course, if you have any prior on extrema, you should inform the fit with such points.

Not a QMC expert but that sounds reasonable. Maybe, @chrisb83 has a better input here. IMO, this seems tangential to the core concepts of random number sampling and is better done in a follow-up PR.

Reply:
Can be a follow up OC.

Reply:
I think there is a misunderstanding...

> Here we are using 30 points to estimate a PDF right? 

No, we choose 30 points where a tangent line will be constructed to derive the hat function (function larger than the pdf). The more points you choose, the closer the hat will be to the pdf and the faster the sampling will be

Reply:
No I think we are roughly talking about the same thing. As, if the problem is then to sample a variable and trying to cover the parameter space in the best possible way. That's what's QMC is for, good space filling properties. OC, if you have prior like the position of some modes and such, you can do smarter tailored sampling. But without, QMC sampling would be at least as good as MC samples.